### ISSUE TITLE
How to specify desired accuracy / resolution of data?

### ISSUE THREAD

**Issue by pes10k** [w3c/geolocation-api#49]
The spec currently says that

[Section 7.2 Requirements](https://w3c.github.io/geolocation-api/#requirements_section)
```
The Geolocation API MUST allow an application to specify a desired accuracy level of the location information.
```

This is a terrific idea, and enables all sorts of good practices (i might just want to know what city someone lives, but not their street, etc, for ethical, safety and legal reasons).

However, I don't see a way to do this in the API.  I see the `enableHighAccuracy` flag, but i) this is a boolean option, and ii) its not clear to the caller what this will mean, or what the resolution of `true` or `false` will be here.  It would be a terrific to have a way of allowing sites to say "no more accurate than…" or "only city level" or "only use IP based location" etc.  

Basically, the ask here to allow sites to blind themselves from non-needed location data

**Comment by marcoscaceres** [comment]
Note, although useful, we are updating a REC to reflect updated implementations, not add new features. This is probably out of scope for now.

**Comment by pes10k** [comment]
Yep!  Didn't mean this one as any kind of blocking issue.  Definitely feature request level issue.

Though (and its a super minor thing) i think the text in the requirements section about "specify a desired accuracy level" seems a bit at odds with the rest of the document.  But any change requested here would be text-tweak level, not feature request level

**Comment by anssiko** [comment]
As @marcoscaceres rightly points out, new feature proposals beyond editorial text tweaking are out of scope for the Geolocation API since this spec is in maintenance mode. Maintenance includes hardening this quite old API for privacy and security to the extend we can without breaking existing content.

For new feature discussion, there's another API called [Geolocation Sensor](https://w3c.github.io/geolocation-sensor/). Please see its issues labeled [geo-accuracy](https://github.com/w3c/geolocation-sensor/issues?q=label%3Ageo-accuracy+) and submit this feedback there. See also [all open issues](https://github.com/w3c/geolocation-sensor/issues). 

Related, and probably of interest to @pes10k as well:

Feedback is also welcome on the [new use cases](https://w3c.github.io/geolocation-sensor/#use-cases) that motivate every new geolocation feature under consideration. Out of these use cases, especially the background operations are highly demanded by web developers but at the same time they are also the most tough ones to solve in a privacy-preserving manner and thus have not advanced beyond use case solicitation at this point. We're hopeful we'll get there eventually, but we don't yet know what API shape and form will get us there. We think, however, that retrofitting the Geolocation API with new features for use cases such as background operation is not a viable option due to its aging design.

**Comment by pes10k** [comment]
Thank you both for your notes above.  I'm holding off adding comments to [Geolocation Sensor](https://w3c.github.io/geolocation-sensor/) since my understanding is the relevant charter has objections and outstanding issues (especially regarding geolocation).  If those get resolved, i'll happily move this issue over there.

**Comment by anssiko** [comment]
@pes10k wrote:

>Thank you both for your notes above. I'm holding off adding comments to Geolocation Sensor since my understanding is the relevant WG charter has objections (including regarding geolocation). If those get resolved, or if ive misunderstood the situation, i'll happily move this issue over there.

@pes10k I think you may have slightly misunderstood how the W3C Process works. Let me try to clarify.

First, The Director has not announced any changes regarding this WG or the Geolocation Sensor work, so please feel free to submit your feedback to the https://github.com/w3c/geolocation-sensor repo where the context for your proposal is. In particular, please review the issues labeled with [geo-accuracy](https://github.com/w3c/geolocation-sensor/issues?q=label%3Ageo-accuracy+) and chime in.

Second, the WG is not aware of any technical arguments against the Geolocation Sensor spec that'd suggest changes to the direction. The WG acknowledges the historical context and that the problem is not easy to solve, but that's not a valid reason to disinvest. This, especially given the strong demand for the new geolocation features from the web developer community. As per the WG's commitment, we solve also this problem in a secure and privacy-preserving manner. As a WG, we also benefit from horizontal review from experts in privacy, security, others. Of course, we're lucky to have our resident privacy experts in this WG who are scrutinizing the WG's work closely. And we have you watching us!

Last but not least, the fact that you opened this issue adds to the pile of evidence we need to keep the Geolocation Sensor as its own separate WG deliverable as to allow discussion and development of new features for geolocation without interfering with the Geolocation API maintenance effort that is also important, but has its own timelines and priorities. These two APIs coexist.

I'll ping @samuelweiler to ensure this valuable feedback from @pes10k is considered by the W3M.

**Comment by pes10k** [comment]
@anssiko I appreciate your comments above, but there as been, i believe, at least a few AC comments asking the "Devices and Sensors Working Group" to cease work on the geolocation spec, and for it to stay in this WG.  I'd like to wait for the resolution of that conversation before moving comments over.  Either way, i will make sure this issue winds up in one (or both) repos

**Comment by anssiko** [comment]
@pes10k wrote:

>but there as been, i believe, at least a few AC comments asking the "Devices and Sensors Working Group" to cease work on the geolocation spec, and for it to stay in this WG

Given "Devices and Sensors Working Group" == "this WG", I feel obligated to clarify one more thing: both the Geolocation API ([since March 2019](https://www.w3.org/2019/03/devices-sensors-wg-charter.html)) and Geolocation Sensor ([since June 2018](https://www.w3.org/2018/06/devices-sensors-wg-charter.html)) are deliverables of the Devices and Sensors Working Group.

Geolocation Working Group that produced an initial version of the Geolocation API [closed in May 2017](https://lists.w3.org/Archives/Public/public-geolocation/2017May/0002.html) leaving that spec in an unmaintained state. Devices and Sensors Working Group was concerned about that and adopted the Geolocation API spec to continue its maintenance in parallel to the new feature work in the Geolocation Sensor spec it had started earlier.

**Comment by samuelweiler** [comment]
I'm going to take this thread in a different direction: it seems odd to me that this spec lists some requirements (with 2119 MUSTs, even) that are then not met - or very weakly met (e.g. with a single bit?!?!?).  A casual reviewer might take solace from reading the requirements, even though they don't reflect the reality of the spec.

I suggest adding a note to this requirements section saying that the spec does not meet requirement #7 (or does a poor job of meeting it).  Ideally, the WG would consider each of the other items also.

I think this suggestion falls into the realm of text tweaking rather than new features.

**Comment by samuelweiler** [comment]
Our security reviewer suggested this feature in https://github.com/w3c/geolocation-api/issues/57 without noticing what I observed on [July 2nd.](https://github.com/w3c/geolocation-api/issues/49#issuecomment-653037317)

I'm flagging this for resolution.

**Comment by marcoscaceres** [comment]
Closed via b1fac55c0f992d7e64de84b15a8203209d2f8029

**Comment by marcoscaceres** [comment]
oops, closed the wrong bug.

**Comment by marcoscaceres** [comment]
Clarified that, “And through enableHighAccuracy, supports requesting "high accuracy" position data, though the request can be ignored by the user agent.” 

In the introduction. 
https://w3c.github.io/geolocation-api/#introduction

**Comment by samuelweiler** [comment]
Looking back on this, it looks like [the requirements text](https://pr-preview.s3.amazonaws.com/w3c/geolocation-api/63/c0381e2...3518ac5.html#usecases) I [comment on above](https://github.com/w3c/geolocation-api/issues/49#issuecomment-653037317) has been entirely removed.

That's somewhat unsatisfying.  As @pes10k writes, the requirement spoke to a useful feature; "This is a terrific idea, and enables all sorts of good practices (i might just want to know what city someone lives, but not their street, etc, for ethical, safety and legal reasons)."  

I think the WG wants to not add any functionality in this space?  Reopening this to confirm the WG intent.  Should we file this as a feature request on a future spec?

**Comment by anssiko** [comment]
>I think the WG wants to not add any functionality in this space? Reopening this to confirm the WG intent. Should we file this as a feature request on a future spec?

Per https://www.w3.org/2020/11/das-wg-charter.html the Geolocation API is in maintenance and as such the focus of the WG is to specify what has been implemented and move the Geolocation API with that feature set toward REC. @marcoscaceres has been driving this effort as an editor, working with implementers to ensure what is specified is also implemented.

New features and functionality have been discussed in the context of the Geolocation Sensor spec based on the Generic Sensor API. Please review the issues labeled with [geo-accuracy](https://github.com/w3c/geolocation-sensor/issues?q=label%3Ageo-accuracy+) and chime in there.

**Comment by marcoscaceres** [comment]
Echoing what @anssiko said. This is probably in scope for the updated REC. But definately something we can explore after, as this will be an "updatable web". It would need implementer buy-in tho.

**Comment by marcoscaceres** [comment]
@samuelweiler, please see https://github.com/w3c/geolocation-api/issues/49#issuecomment-993312680 and my comment above. We can keep this issue open, but would like a resolution that it will either be part of Geolocation Sensor or something we work on after REC.

**Comment by pes10k** [comment]
Just following up here after figuring out who PING side should follow up.  PING-side, we're fine closing this as long as the WG agrees to take it up post REC

**Comment by marcoscaceres** [comment]
@pes10k, yep, we agree to take it up later. Let's leave it open.

@samuelweiler, does it need to be resolved over on some other (security) tracker? (i.e., the "security-needs-resolution" label)

**Comment by samuelweiler** [comment]
Thanks, @marcoscaceres.  I did some tracker cleanup.
